config/config_params_test.go

```diff
@@ -74,8 +75,8 @@ var _ = Describe("FelixConfig vs ConfigParams parity", func() {
 	}
 
 	BeforeEach(func() {
-		fcFields = fieldsByName(v3.FelixConfigurationSpec{})
-		cpFields = fieldsByName(Config{})
+		fcFields = exportedFieldsByName(v3.FelixConfigurationSpec{})
```

Comment:
I felt we didn't want to compare any unexported fields, i.e. loadClientConfigFromEnvironment, as the internals of an object shouldn't should matter to the outside (I know the rules on the "internals of an object" are a bit different golang, with respect to packages, but for the config struct I think the rule holds)

Reply:
Yes, but we already have a solution for that, i.e. `cpFieldsToIgnore`.  I think it would be much simpler just to add loadClientConfigFromEnvironment to that.

Reply:
I did see this, but I added that because I couldn't see a reason why we would want to make sure that unexported fields are the same, because every time somebody adds an unexported field we would have to update this test, so this seemed simpler to me in the long run.

Reply:
I'm afraid not.  It's unnecessary code churn - so will impact e.g. open source -> private merges.  And it's just _more_ code in config_parms_test.go, so makes it harder whenever someone needs to read and understand that again.  Please revert.

Reply:
Reverting